skbeam/core/utils.py

```diff
@@ -1271,3 +1271,36 @@ def bin_grid(image, r_array, pixel_sizes, statistic='mean', mask=None,
     bin_centers = bin_edges_to_centers(bin_edge)
 
     return bin_centers, int_stat
+
+
+def time_binning(time_diff, ff=2):
+    """
+    Bin the time differences into all possible bins
+
+    Parameters
+    ----------
+    time_diff : list
+        time difference from previous image for each image as a list
+    ff : int, optional
+        desired float point format
+
+    Returns
+    -------
+    time_bin : list
+        time bin for each delay time according to time bins
+    all_lags : array
+        all possible time lags from the time difference
+
+    """
+    time_bins = []
+    all_lags = []
+    ff_p = '.' + str(ff) + 'f'
+    for c, t in enumerate(time_diff):
+        stamp = 0
+        time_b = []
+        for item in (time_diff[:c+1][::-1]):
+            time_b.append(float(format(stamp, ff_p)))
```

Comment:
In general, appending a numpy array to a Python list one and a time and converting floats to strings in order to perform a numerical operation is a red flag that there must be a better way. This will become very slow for long time series. There is, for example, `np.round` which would accomplish this in a vectorized way.